### PR TITLE
Create presigned urls

### PR DESCRIPTION
**Pull request by carlos-alves-bc**
## :pencil: Changelog:

- :sparkles: Features:
    - Create presigned url functions: `get_presigned_download_url`, `get_presigned_post_url` and `get_presigned_url`

### CODE REVIEW

File: tests/unit/objects/test_presigned_url.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        assert url is not None
+        assert response.status_code == 200
+
+    def test_list_bucket_objects_with_presigned_url(self, s3_client):
+        with create_bucket(s3_client, BUCKET_NAME, keys_paths=[(self.key, FILENAME)]):
+            url = get_presigned_url(client_method='list_objects', method_parameters={'Bucket': BUCKET_NAME})
+            response = requests.get(url)
+
+        assert response.status_code == 200
```
Comment (tomassosorio) [inline tests/unit/objects/test_presigned_url.py:46]:
```suggestion

        assert url is not None
        assert response.status_code == 200
```

File: tests/unit/objects/test_presigned_url.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    fn_test = FILENAME + ".tests"
+    key = "prefix/object"
+
+    def test_download_nonexisting_object_with_presigned_url(self, s3_client):
+        with create_bucket(s3_client, BUCKET_NAME), pytest.raises(requests.exceptions.HTTPError):
+            url = get_presigned_download_url(bucket=BUCKET_NAME, key=self.key)
+            response = requests.get(url)
+            response.raise_for_status()
+
+            assert response.status_code == 404
```
Comment (tomassosorio) [inline tests/unit/objects/test_presigned_url.py:31]:
```suggestion

            assert url is None
            assert response.status_code == 404
```

File: s3_tools/objects/presigned_url.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
+"""Create S3 bucket objects presigned url."""
+from typing import (
+    Dict,
+    Optional
+)
+
+import boto3
+
+
+def get_presigned_url(
```
[the remark this answers is not in the document]
Comment (carlos-alves-bc) [reply s3_tools/objects/presigned_url.py:10]: I am not sseing a better name for this. Presigned url is a specific feature in AWS S3 and it is called like this in official documentation 😅

File: tests/unit/objects/test_presigned_url.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+)
+
+from tests.unit.conftest import (
+    BUCKET_NAME,
+    FILENAME,
+    create_bucket
+)
+
+
+class TestPresignedUrl:
```
[the remark this answers is not in the document]
Comment (cmalves) [reply tests/unit/objects/test_presigned_url.py:21]: That is not the pattern we are using in the tests of the other modules 😕 We can talk later and see if we want to split this and other module tests

**Review by tomassosorio — APPROVED** [review]

**Review by FerrariDG — APPROVED** [review]

**Pull request merged**